test/parse.jl

```diff
@@ -226,6 +226,27 @@ end
 # error throwing branch from #10560
 @test_throws ArgumentError Base.tryparse_internal(Bool, "foo", 1, 2, 10, true)
 
+# issue #16594
+@test length(:(@test 1 +
+                     1 == 2).args) == 3
+@test [1 +
+       1] == [2]
+@test [1 +1] == [1 1]
+@test length(:(@x 1 +1 -1).args) == 5
+@test length(:(@x 1 + 1 -1).args) == 4
+@test length(:(@x 1 + 1 - 1).args) == 3
+@test length(:(@x 1 +
+                  1 -
+                  1).args) == 3
+@test length(:(@x 1 +
+                  1 +
+                  1).args) == 3
```

Comment:
I would prefer these tests to check more than just the number of arguments.

Reply:
Yeah, just check Expr equality?